presto-main/src/main/java/io/prestosql/type/TypeCoercion.java

```diff
@@ -437,6 +437,9 @@ private TypeCompatibility typeCompatibilityForCovariantParametrizedType(Type fro
             case StandardTypes.CHAR: {
                 switch (resultTypeBase) {
                     case StandardTypes.VARCHAR:
+                        // CHAR could be coercible to VARCHAR, but they cannot be both coercible to each other.
+                        // VARCHAR to CHAR coercion provides natural semantics when comparing VARCHAR literals to CHAR columns.
+                        // WITH CHAR to VARCHAR coercion one would need to pad literals with spaces: char_column_len_5 = 'abc  ', so we would not run unmodified TPC-DS queries.
                         return Optional.empty();
```

Comment:
This line had and still has a warning (DuplicateBranchesInSwitch).
I am tempted to "ignore this globally" (in my IDE and in https://github.com/prestosql/presto/#additional-ide-configuration)

cc @electrum